boot/zephyr/main.c

```diff
@@ -44,6 +44,10 @@ const struct boot_uart_funcs boot_funcs = {
 #include <usb/class/usb_dfu.h>
 #endif
 
+#ifdef CONFIG_BOARD_NRF9160_PCA20035
```

Comment:
This looks more like `nrf temphack` than `nrf noup` to me.

Do the existing application specific hooks in `<power/power.h>` not allow this to be done in a more generic way? I can't see why we want this to be just for one particular upstream board, since any nRF-derived board (including customer designs) could want this same feature, no?

Reply:
I do not find any thing in power.h for this use case, but it can most likely be done more generic for the the nRF devices. Something like below would probably be safe and generic?

#ifdef CONFIG_SOC_FAMILY_NRF
#if NRF_POWER_HAS_RESETREAS
....
#endif 
#endif

Reply:
That's better! How about going all the way with something like this?

```c
#ifdef CONFIG_SOC_FAMILY_NRF
#include <hal/nrf_power.h>

static inline bool boot_skip_serial_recovery()
{
	u32_t rr = nrf_power_resetreas_get();

	return !(rr == 0 || (rr & NRF_POWER_RESETREAS_RESETPIN_MASK));
}
#else
static inline bool boot_skip_serial_recovery()
{
	return false;
}
#endif

/* ... */

if (detect_value == CONFIG_BOOT_SERIAL_DETECT_PIN_VAL && \
	!boot_skip_serial_recovery()) {

}
```

I bet that would be relatively easy to upstream into MCUboot, since 1. zephyr doesn't have a generic API for this you could find and 2. the vendor-specific portions are hidden away a bit more from the application code

Reply:
ok. I did not see the comment before I updated the PR with a variant based on my first commit. I will try out what you propose when back in office tomorrow.

Reply:
Sounds good, thanks! I think you might as well submit it upstream if the approach looks OK to you so we don't have to carry the downstream patch.